src/busterbunny.js

```diff
@@ -70,14 +70,25 @@ module.exports = (function() {
         };
 
         self.raiseEvents = function (eventId, event, options, afterRaised) {
-            _stats.queuedEventsToRaise++;
-            if (options instanceof Function) {
+            if(options instanceof Function && afterRaised)
+                throw new Error('the third argument must be an object when a fourth argument is provided.');
+
+            if (options instanceof Function && !afterRaised) {
                 afterRaised = options;
                 options = null;
             }
 
-            var eventData = self.encoder.encodeEvent(event);
-            publishOrQueue(config.exchange, eventId, eventData, options, afterRaised);
+            if(!(afterRaised instanceof Function))
+                throw new Error('the argument provided for callback is not a Function');
+
+            if(typeof eventId !== 'string') {
+                afterRaised(new Error('eventId is required and must be a string'));
```

Comment:
Maybe I'm missing something, but I feel this should throw as well. If eventId is not a string, that is clearly a violation of what is expected, and should therefore, fail similarly to other such violations